File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/chat/types/PersonalChatMessageModel.ts
import type { MessageStatus } from './enums/MessageStatus';

export type PersonalChatMessageModel = {
    id: number;
    username: string;
    message: string;
    time: Date;
    status: MessageStatus;
    type: number;
    markedType: number;
    isEdited: boolean;
    personalChatId: number;
    appUserId: string;
}

File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/Chat/PersonalChatMessageModel.cs
namespace CombatAnalysis.EnhancedWebApp.Server.Models.Chat;

public record PersonalChatMessageModel(
    int Id,
    string Username,
    string Message,
    DateTimeOffset Time,
    int Status,
    int Type,
    int MarkedType,
    bool IsEdited,
    int PersonalChatId,
    string AppUserId
    );